libs/pulumi-lambda/src/lib/metric-alarms.ts

```diff
@@ -0,0 +1,289 @@
+import * as pulumi from '@pulumi/pulumi'
+import * as aws from '@pulumi/aws'
+
+export const defaultThresholds = {
+    avgDurationMs: 3000,
+    maxDurationMs: 15000,
+    errorRatePercent: 2,
+    throttles: 1,
+    concurrents: 450,
+}
+
+export interface Thresholds {
+    /**
+     * Alert if the average duration is greater than this
+     * Default: 3000ms (3 seconds)
+     */
+    avgDurationMs?: pulumi.Input<number>
+    /**
+     * Alert if the max duration is greater than this
+     * Default: not used (undefined)
+     */
+    maxDurationMs?: pulumi.Input<number>
+    /**
+     * Alert if the max duration is greater than this
+     * Default: not used (undefined)
+     */
+    timeoutMs?: pulumi.Input<number>
+    /**
+     * Alert if the error rate exceeds this
+     * (default: 2 percent)
+     */
+    errorRatePercent?: pulumi.Input<number>
+    /**
+     * Throttles are bad mkay
+     * default: 1 throttle in the last 5 minutes
+     */
+    throttles?: pulumi.Input<number>
+    /**
+     * Alert if concurrent executions exceeds this
+     * This has a cost implication, but it can also
+     * cause throttles if it rises too fast
+     * default: more than 450 concurrents at any
+     * time in the past 5 minutes
+     */
+    concurrents?: pulumi.Input<number>
+}
+
+/**
+ * Provides common metric alarms for a lambda function
+ */
+export class MetricAlarms extends pulumi.ComponentResource {
```

Comment:
Another alternative is to export a bunch of different alarms so it's more composible.

ie.

`new LambdaErrorRateAlarm(...)` etc

Reply:
I like this idea, but there's some value in providing an opinionated set of recommended alarms that should probably be enabled for all lambda functions.

Maybe it could also export a RecommendedAlarms class, that just wraps a few of the recommended alarms?

I've updated the PR; let me know what you think of this idea.